runners/direct-java/src/main/java/org/apache/beam/runners/direct/EvaluationContext.java

```diff
@@ -183,6 +188,15 @@ public void initialize(
         committedResult.getUnprocessedInputs().orElse(null),
         committedResult.getOutputs(),
         result.getWatermarkHold());
+
+    // Callback and requested bundle finalizations
+    for (InMemoryBundleFinalizer.Finalization finalization : result.getBundleFinalizations()) {
+      try {
+        finalization.getCallback().onBundleSuccess();
+      } catch (Exception e) {
```

Comment:
Is is possible to make this Exception more specific? Or that does not add more values?

Reply:
This exception is to catch user thrown exceptions since the bundle finalizing callback is user written code this is why its so wide and it can't be narrowed.